File: Easy/2026-08-11-2996-Smallest-Missing-Integer-Greater-Than-Sequential-Prefix-Sum/solution.java
class Solution {
    public int smallestMissingGreaterThanPrefixSum(int[] nums) {
        int maxLength = 0;
        int currentLength = 1;
        int prefixSum = nums[0];
        
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i - 1] + 1) {
                currentLength++;
                prefixSum += nums[i];
            } else {
                maxLength = Math.max(maxLength, currentLength);
                currentLength = 1;
                prefixSum = nums[i];
            }
        }
        maxLength = Math.max(maxLength, currentLength);
        
        int longestPrefixSum = 0;
        for (int i = 0; i < maxLength; i++) {
            longestPrefixSum += nums[i];
        }
        
        boolean[] present = new boolean[51];
        for (int num : nums) {
            present[num] = true;
        }
        
        for (int x = longestPrefixSum; x <= 50; x++) {
            if (!present[x]) {
                return x;
            }
        }
        
        return 51; // In case all numbers from longestPrefixSum to 50 are present
    }
}

File: Easy/2026-08-11-2996-Smallest-Missing-Integer-Greater-Than-Sequential-Prefix-Sum/solution.py
class Solution:
    def smallestMissingInteger(self, nums: List[int]) -> int:
        n = len(nums)
        longest_prefix_sum = 0
        current_sum = 0
        
        for i in range(n):
            if i == 0 or nums[i] == nums[i - 1] + 1:
                current_sum += nums[i]
                longest_prefix_sum = current_sum
            else:
                break
        
        missing = longest_prefix_sum
        while missing in nums:
            missing += 1
        
        return missing